src/threadpool.c

```diff
@@ -55,6 +58,11 @@ static void worker(void* arg) {
 
   (void) arg;
 
+  uv_mutex_lock(&threads_started_mutex);
+  ++threads_started;
+  uv_cond_signal(&threads_started_cond);
+  uv_mutex_unlock(&threads_started_mutex);
```

Comment:
This is a hand-rolled semaphore.
The code would be more readable if a semaphore were used as described [here](https://github.com/libuv/libuv/pull/1639#issuecomment-348071153).

Is this actually [cheaper](https://github.com/libuv/libuv/pull/1639#issuecomment-348142522) than using a uv_sem_t?